testing/mock.ts

```diff
@@ -1074,8 +1082,8 @@ export function returnsThis<
 // deno-lint-ignore no-explicit-any
 export function returnsArg<Arg, Self = any>(
   idx: number,
-): (this: Self, ...args: Arg[]) => Arg {
-  return function (...args: Arg[]): Arg {
+): (this: Self, ...args: Arg[]) => Arg | undefined {
+  return function (...args: Arg[]): Arg | undefined {
```

Comment:
This is a change to the public API.

Its always been the case that this could return `undefined` if the index is outside the array bounds, now the type just reflects reality. Alternativly I could add a check and throw which would keep the type consistent but change the runtime behaviour.

Reply:
I think throwing if the index is outside of the argument bounds is a better idea.

Reply:
Done